### PR TITLE
Refactor Password Modal

### PR DESCRIPTION
**Pull request by NovacAndrei**
#527 
#531 
#532

### CODE REVIEW

File: src/screens/view-wallet-mnemonic/view-wallet-mnemonic.tsx [hunk 1]
```diff
@@ -49,14 +48,15 @@ export class ViewWalletMnemonicScreenComponent extends React.Component<
...
-                this.populateMnemonic(password);
-            })
-            .catch(() => {
-                this.props.navigation.goBack(null);
-            });
+        PasswordModal.getPassword(
+            translate('Password.pinTitleUnlock'),
+            translate('Password.subtitleMnemonic')
+        ).then(password => this.populateMnemonic(password));
+
```
Comment (georgeburduhos) [inline src/screens/view-wallet-mnemonic/view-wallet-mnemonic.tsx]: if it fails? This screen should decide if we need to ask the password or not, not the previous

File: src/screens/token/components/account-settings/account-settings.tsx [hunk 2]
```diff
@@ -55,8 +53,9 @@ export class AccountSettingsComponent extends React.Component<IProps & IExternal
         };
     }
 
-    public revealPrivateKey = async () => {
-        const password = await this.passwordModal.requestPassword();
+    public async revealPrivateKey() {
+        // TODO: not working because of modal over modal
```
Comment (georgeburduhos) [inline src/screens/token/components/account-settings/account-settings.tsx]: why is there a todo here? - this is not working?

File: src/core/i18n/translation/locales/en.ts [hunk 3]
```diff
@@ -158,7 +160,19 @@ export const translation: ITranslations = {
             changePinSubtitle: 'Your current PIN is required in order to change it',
             recoverWalletPinSubtitle: 'Your PIN is required\nto recover your Moonlet wallet',
             connectHardwareWalletPinSubtitle:
-                'Your PIN is required\nto connect your hardware wallet'
+                'Your PIN is required\nto connect your hardware wallet',
+            moonletDisabled: 'Moonlet is disabled',
+            disabledDetails: 'try again in {{duration}} {{measurement}}'
+        },
+        Time: {
+            second: 'second',
```
Comment (georgeburduhos) [inline src/core/i18n/translation/locales/en.ts]: are all these actually used?

File: src/components/password-modal/password-modal-component.tsx [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+    newPassword: string;
+    currentStep: ScreenStep;
+    errorMessage: string;
+    enableBiometryAuth: boolean;
+    countdownListenerTime: number;
+    allowBackButton: boolean;
+    showAttempts: boolean;
+}
+
+const EMPTY_STRING = ' ';
```
Comment (georgeburduhos) [inline src/components/password-modal/password-modal-component.tsx]: EMPTY_STRING ? - why not just use ''

File: src/components/password-modal/password-modal-component.tsx [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+    }
+
+    public createPassword(subtitle?: string) {
+        this.resultDeferred = new Deferred();
+        this.clearErrorMessage();
+
+        this.setState({
+            visible: true,
+            title: translate('Password.setupPinTitle'),
+            subtitle,
```
Comment (georgeburduhos) [inline src/components/password-modal/password-modal-component.tsx]: shouldn't be a fallback subtitle?

File: src/core/i18n/translation/locales/en.ts (hunk 3 above)
Comment (NovacAndrei) [reply src/core/i18n/translation/locales/en.ts]: Yes, all of them are used
image

File: src/components/password-modal/components/password-pin/password-pin.tsx [hunk 6]
```diff
@@ -11,7 +11,17 @@ import { Icon } from '../../../icon';
...
-import { normalize } from '../../../../styles/dimensions';
+import { normalize, ICON_SIZE } from '../../../../styles/dimensions';
+import { SafeAreaView } from 'react-navigation';
+
+const digitsLayout = [
+    [1, 2, 3],
+    [4, 5, 6],
+    [7, 8, 9]
+];
+const ZERO = 0;
```
[the remark this answers is not in the document]
Comment (NovacAndrei) [reply src/components/password-modal/components/password-pin/password-pin.tsx]: image

File: src/screens/token/components/account-settings/account-settings.tsx (hunk 2 above)
Comment (NovacAndrei) [reply src/screens/token/components/account-settings/account-settings.tsx]: This is not working,  I am going to fix this in another task

File: src/components/password-modal/components/password-pin/password-pin.tsx [hunk 7]
```diff
@@ -72,7 +71,7 @@ export class PasswordPinComponent extends React.Component<
 
     public componentDidUpdate(prevProps: IExternalProps) {
         if (this.props.errorMessage !== prevProps.errorMessage) {
-            if (this.props.errorMessage !== EMPTY_STRING) {
+            if (this.props.errorMessage !== '') {
```
Comment (krisboit) [inline src/components/password-modal/components/password-pin/password-pin.tsx]: it's more safe to do `if (this.props.errorMessage) {`

File: src/components/password-modal/password-modal-component.tsx [hunk 8]
```diff
@@ -194,28 +192,25 @@ export class PasswordModalComponent extends React.Component<
...
 
     private clearErrorMessage() {
-        this.setState({ errorMessage: EMPTY_STRING });
+        this.setState({ errorMessage: '' });
     }
 
     private handlePasswordAttempts() {
+        const failedLoginBlocking = Object.keys(FAILED_LOGIN_BLOCKING)
+            .concat(String(RESET_APP_FAILED_LOGINS))
+            .reverse();
```
Comment (krisboit) [inline src/components/password-modal/password-modal-component.tsx]: to be sure the array is ordered descending you should use: `.sort((a,b) => b-a)`

**Pull request merged**